### PR TITLE
Update webhook.go

### PR DESCRIPTION
**Pull request by bryanpaget**
add validation in the filer injector logic to gracefully handle some potential problems that may come up

#### Jira
- https://jirab.statcan.ca/browse/BTIS-418
- https://jirab.statcan.ca/browse/BTIS-447

### CODE REVIEW

File: cmd/webhook.go [hunk 1]
```diff
@@ -248,7 +249,15 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
...
 
-			// TODO: dd validation and graceful exit if bucketMount is empty
+			// S3_URL, S3_ACCESS, and S3_SECRET are essential
+			s3Url := string(secretList.Items[sec].Data["S3_URL"])
+			s3Access := string(secretList.Items[sec].Data["S3_ACCESS"])
+			s3Secret := string(secretList.Items[sec].Data["S3_SECRET"])
+
+			// Validation: Ensure bucketMount, S3_URL, S3_ACCESS, and S3_SECRET are present and not empty
+			if bucketMount == "" || s3Url == "" || s3Access == "" || s3Secret == "" {
+				continue // Skip this secret if any of the necessary values are empty
```
Comment (Souheil-Yazji) [inline cmd/webhook.go:259]: Should at the very least leave a trace of skipped secret (log.printf)

File: cmd/webhook.go [hunk 2]
```diff
@@ -264,21 +273,27 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 				filerBucketName = filerBucketName + "-" + limitDeepestDirName
 			}
 
+			// Validation: Ensure container and volume names follow the correct naming convention
+			validNameRegex := regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
+			if !validNameRegex.MatchString(filerBucketName) {
+				continue // Skip this secret if the generated name is invalid
```
Comment (Souheil-Yazji) [inline cmd/webhook.go:279]: graceful trace

**Review by Souheil-Yazji — CHANGES_REQUESTED** [review]
good! See comments

File: cmd/webhook.go (hunk 1 above)
Comment (bryanpaget) [reply cmd/webhook.go:261]: I added logging with `warningLogger.Printf("Skipping secret %s in namespace %s: generated name %s is invalid according to the naming convention"...`

File: cmd/webhook.go (hunk 2 above)
Comment (bryanpaget) [reply cmd/webhook.go:278]: I added logging with `warningLogger.Printf("Skipping secret %s in namespace %s: generated name %s is invalid according to the naming convention"...`

**Review by Souheil-Yazji — APPROVED** [review]

File: cmd/webhook.go [hunk 3]
```diff
@@ -289,17 +301,58 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
...
+	return name
+}
+
+// Function to ensure name uniqueness by appending an integer if the name already exists
+func ensureUniqueName(baseName string, existingNames []string) string {
+	// Check if the name is already in the list of existing names
+	if slices.Contains(existingNames, baseName) {
+		// Append an integer to make the name unique
+		for i := 1; ; i++ {
+			newName := fmt.Sprintf("%s-%d", baseName, i)
```
Comment (mathis-marcotte) [inline cmd/webhook.go:339]: Isn't this function used to print formated strings? Does it actually return a string? Wouldnt it be more clear to do it like it used to be, like `filerBucketName = filerBucketName + "-" + strconv.Itoa(len(filerBucketList)+1)` , so in your case something like `newName := baseName + "-" + strconv.Itoa(i)`

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]

File: cmd/webhook.go (hunk 3 above)
Comment (mathis-marcotte) [reply cmd/webhook.go:339]: sprintf works.

**Review by mathis-marcotte — APPROVED** [review]

**Pull request merged**